scripts/resolve-includes.groovy

```diff
@@ -157,9 +160,50 @@ private void copyInvokerPropertiesToProjects(List<String> projects) {
     }
 }
 
+private List<String> resolveProjectExplicitFiltering(ArrayList<String> leafProjects) {
+    List<String> projects = mavenProjectExplicitFiltering.split(',')
+    List<String> filteredLeafProjects = new ArrayList<>()
+    List<String> filteredOutLeafProjects = new ArrayList<>()
+    String filteringTYpe = mavenProjectExplicitFilteringType
+
+    def logFile = "filter-projects.log"
+    log.info("Starting explicit filtering projects resolution, results will be in ${Path.of(logdir).resolve(logFile).toString()}")
+
+    if (projects && !projects.empty) {
+        for (int i = 0; i < leafProjects.size(); i++) {
```

Comment:
Adds the filtered elements to the lists (filter in / filter out). It could be only one list, but I'm focusing on legibility.